oecophylla/taxonomy/taxonomy.rule

```diff
@@ -221,34 +219,46 @@ rule taxonomy_shogun:
     output:
         taxon_counts = taxonomy_dir + "{sample}/shogun/{sample}.taxon_counts.tsv"
     params:
-        shogun = config['params']['shogun'],
-        shogun_env = config['envs']['shogun']
-    threads:
-        2
+        env = config['envs']['shogun'],
+        db = config['params']['shogun']['db'],
+        threads = config['params']['shogun']['threads'],
+        aligner = config['params']['shogun']['aligner'],
+        level = config['params']['shogun']['level']
     log:
         taxonomy_dir + "logs/taxonomy_shogun.sample=[{sample}].log"
     benchmark:
         "benchmarks/taxonomy/taxonomy_shogun.sample=[{sample}].txt"
     run:
+        aln2ext = {'utree': 'tsv', 'burst': 'b6', 'bowtie2': 'sam'}
+        ext = aln2ext[params['aligner']]
         with tempfile.TemporaryDirectory(dir=find_local_scratch(TMP_DIR_ROOT)) as temp_dir:
             shell("""
-                  set +u; {params.shogun_env}; set -u
+                  set +u; {params.env}; set -u
 
                   # convert and merge fastq's into fasta
                   seqtk seq -A {input.forward} > {temp_dir}/{wildcards.sample}.fna
                   seqtk seq -A {input.reverse} >> {temp_dir}/{wildcards.sample}.fna
 
-                  # run shogun
-                  shogun_utree_lca.py {params.shogun} \
-                  --threads {threads} \
-                  --input {temp_dir} \
+                  # map reads to reference database
+                  shogun align \
+                  --aligner {params.aligner} \
+                  --threads {params.threads} \
+                  --database {params.db} \
+                  --input {temp_dir}/{wildcards.sample}.fna \
                   --output {temp_dir} \
                   2> {log} 1>&2
 
-                  # parse output
-                  echo '#'SampleID$'\t'{wildcards.sample} > {output.taxon_counts}
-                  cat {temp_dir}/taxon_counts.csv | \
-                  tail -n+2 | tr "," "\\t" >> {output.taxon_counts}
+                  # build taxonomic profile based on read map
+                  shogun assign_taxonomy \
+                  --aligner {params.aligner} \
+                  --database {params.db} \
+                  --input {temp_dir}/alignment.{params.aligner}.{ext} \
+                  --output {output.taxon_counts} \
+                  2> {log} 1>&2
+
+                  # clean up
```

Comment:
temp dir should get automatically deleted after job finishes. No need to clean up?

Reply:
Oh I see. Thanks for letting me know!